java-client/src/main/java/org/opensearch/client/transport/aws/AwsSdk2Transport.java

```diff
@@ -694,7 +694,11 @@ private static Exception extractAndWrapCause(Exception exception) {
             if (t instanceof Error) {
                 throw (Error) t;
             }
-            exception = (Exception) t;
+            if (t instanceof Exception) {
+                exception = (Exception) t;
+            } else {
+                exception = new RuntimeException("Unexpected throwable caught in execution", t);
```

Comment:
I am wondering if `UndeclaredThrowableException` would be an appropriate exception container in this case?

```suggestion
                exception = new UndeclaredThrowableException(t);
```

Reply:
Thanks. It looks more appropriate to me too!